src/onboarding/Onboarding.js

```diff
@@ -0,0 +1,583 @@
+          }),
+          {}
+        ),
+      }),
+      {}
+    )
+  }
+
+  getInitialDataFromTemplate(template) {
+    if (!Templates.has(template)) {
+      return []
+    }
+    return Templates.get(template).screens.map(({ screen, fields }) => ({
+      screen,
+      data: Object.entries(fields).map(([name, { defaultValue }]) => ({
+        name,
+        value: defaultValue(),
+      })),
+    }))
+  }
+
+  // Return a screen object from a template
+  getTemplateScreen(template, screen) {
+    if (!Templates.has(template)) {
+      return null
+    }
+    return (
+      Templates.get(template).screens.find(
+        screenData => screenData.screen === screen
+      ) || null
+    )
+  }
+
+  // Filters a field value by calling the corresponding filter on the template
+  filterConfigurationValue(template, screen, name, value) {
+    const screenData = this.getTemplateScreen(template, screen)
+    return screenData ? screenData.fields[name].filter(value) : null
+  }
+
+  // Check if the data is valid by calling validateScreen() on the template
+  validateConfigurationScreen(template, screen) {
+    const screenData = this.getTemplateScreen(template, screen)
+    return screenData
+      ? screenData.validateScreen(this.getTemplateData())
+      : false
+  }
+
+  handleStartCreate = () => {
+    this.reset()
+    this.moveStep(1)
+  }
+
+  handleStartRest = () => {
+    // Unset the template and the domain when
+    // the home screen finishes its transition.
+    const { stepIndex } = this.state
+    if (stepIndex === 0) {
+      this.setState({ template: null, domain: '' })
+    }
+  }
+
+  handleTemplateSelect = (template = null) => {
+    this.setState({
+      template,
+      templateData: this.getInitialDataFromTemplate(template),
+    })
+  }
+
+  checkDomain = (
+    domain,
+    domainKey,
+    domainStatusKey,
+    timerKey,
+    invertCheck = false
+  ) => {
+    const { daoBuilder } = this.props
+    const filteredDomain = domain.trim().toLowerCase()
+
+    this.setState({
+      [domainKey]: filteredDomain,
+      [domainStatusKey]: DomainCheckPending,
+    })
+    clearTimeout(this[timerKey])
+
+    // Empty name
+    if (!filteredDomain) {
+      this.setState({ [domainStatusKey]: DomainCheckNone })
+      return
+    }
+
+    const checkName = async () => {
+      try {
+        const available = await daoBuilder.isNameAvailable(filteredDomain)
+
+        // The domain could have changed in the meantime
+        if (filteredDomain === this.state[domainKey]) {
+          const valid = invertCheck ? !available : available
+          const status = valid ? DomainCheckAccepted : DomainCheckRejected
+          this.setState({ [domainStatusKey]: status })
+        }
+      } catch (err) {
+        // Retry every second
+        this[timerKey] = setTimeout(checkName, 1000)
+      }
+    }
+
+    // Check the domain only after the field is not updated during 300ms
+    this[timerKey] = setTimeout(checkName, 300)
```

Comment:
This seems like a good debounce / throttle util that we could pull out later (to also avoid polluting this component).

Reply:
Yes, or we could probably add lodash!